airbyte/_future_cdk/state_writers.py

```diff
@@ -31,9 +72,25 @@ class StdOutStateWriter(StateWriterBase):
     an orchestrator is responsible for saving those state artifacts.
     """
 
-    def write_state(
+    def _write_state(
         self,
         state_message: AirbyteStateMessage,
     ) -> None:
         """Save or 'write' a state artifact."""
         print(state_message.model_dump_json())
+
+
+class NoOpStateWriter(StateWriterBase):
```

Comment:
The no-op state writer also is an in-memory recording, per my note above about state writers now being subclassed from state providers.